subsys/ieee802154/nrf_802154_configurator.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <init.h>
+#include <logging/log.h>
+#include <sys/__assert.h>
+#include <devicetree.h>
+#include <nrf_802154.h>
+#include <nrf_802154_config.h>
+
+LOG_MODULE_REGISTER(nrf_802154_configure);
+
+static uint8_t lna_gain_get(void)
+{
+#if IS_ENABLED(CONFIG_MPSL_FEM_NRF21540_GPIO)
```

Comment:
`IS_ENABLED` can be used with C conditions. This gives advantage of source being checked by compiler and unused things are optimized out.

```
if (IS_ENABLED(CONFIG...)) {
} else if (IS_ENABLED(CONFIG_...)) {
else {
}
```

Reply:
That would simplify and prettify this mess, however those symbols are just not defined in case of a different FEM. Still, I rewritten it to be just a macro definition, not a function.